ujt/transformers.py

```diff
@@ -65,6 +100,116 @@ def apply_views(elements, tag_map, view_list):
         element["classes"] += f" {' '.join(class_list)}"
 
 
+def apply_change_over_time_classes(
+    elements: List[Dict[str, Any]],
+    start_time: dt.datetime,
+    end_time: dt.datetime,
+    sli_type: "SLITypeValue",
+) -> List[Dict[str, Any]]:
+    """Applies classes to elements based on the change in SLIs over the time range.
+
+    Args:
+        elements: A list of cytoscape elements
+        start_time: The start time of the time range to compute aggregate status over.
+        end_time: The end time of the time range to compute aggregate status over.
+        sli_type: The SLI type of interest
+
+    Returns:
+        a A new list of elements with the change over time classes applied.
+    """
+    sli_response = rpc_client.get_slis(
+        start_time=start_time,
+        end_time=end_time,
+        sli_types=[sli_type],
+    )
+    node_name_sli_map = defaultdict(list)
+    for sli in sli_response.slis:
+        node_name_sli_map[sli.node_name].append(sli)
+
+    output_elements = []
+    for element in elements:
+        ujt_id = element["data"]["ujt_id"]
+        if ujt_id not in node_name_sli_map:
+            output_elements.append(element.copy())
+        else:
+            element_copy = element.copy()
+            change_over_time_class = get_change_over_time_class_from_sli_list(
+                node_name_sli_map[ujt_id],
+                start_time,
+                end_time,
+            )
+            element_copy["classes"] += f" {change_over_time_class}"
+            output_elements.append(element_copy)
+
+    return output_elements
```

Comment:
Now we use element.copy() and build a new output elements list, as part of the effort to convert transformers into an immutable style.